django-stubs/test/client.pyi

```diff
@@ -154,8 +155,27 @@ class _RequestFactory(Generic[_T]):
 
 class RequestFactory(_RequestFactory[WSGIRequest]): ...
 
+# A non total duplication of `asgiref.typing.HTTPScope`
 @type_check_only
-class _AsyncRequestFactory(_RequestFactory[_T]): ...
+class HTTPScope(TypedDict, total=False):
```

Comment:
This should be internal. Also maybe let's add some suffix to more clearly distinguish from `HTTPScope`?

```suggestion
class _HTTPScopeNonTotal(TypedDict, total=False):
```

Reply:
I'll update it, but I have to admit that I don't follow on why it's considered internal and should thus be prefixed with `_`. Why is it internal?

Reply:
And to follow up on that. If it's internal I don't see a reason to suffix with `NonTotal`, since it shouldn't be exported to any different module?

Reply:
> I don't follow on why it's considered internal and should thus be prefixed with _. Why is it internal?

Because such a symbol doesn't exist in Django itself. You can't do `from django.test.client import HTTPScope` because that would fail at runtime. (It would only work with a `if TYPE_CHECKING` guard, but that's ugly).

If we *want* to export it for users, it would be preferrable to add it to `django_stubs_ext`.

Reply:
> And to follow up on that. If it's internal I don't see a reason to suffix with NonTotal, since it shouldn't be exported to any different module?

Fair enough

Reply:
I was under the impression that `@type_check_only` helped out with unavailable at runtime? Ref: https://docs.python.org/3/library/typing.html#typing.type_check_only

But no, it doesn't change anything in regards to `if TYPE_CHECKING`

Reply:
> I was under the impression that @type_check_only helped out with unavailable at runtime?

Oh, I had forgotten what it does and had it confused with some other decorators.

But I tried it and it seems that implementations don't follow it in a helpful manner. PyCharm happily suggests imports for such classes, and mypy does not issue any warnings when using them, e.g.:

```python
from django.conf.global_settings import Router
from django.test.utils import TimeKeeperProtocol

print(Router)
print(TimeKeeperProtocol)
```

So currently the `_` prefix is a more foolproof way to indicate stubs-only symbols.

Reply:
I see and agree, let's stick with both